### ISSUE TITLE
Resubscribe on Error should work multiple times

### ISSUE THREAD

**Issue by peterreisz** [ngrx/platform#2023]
## Minimal reproduction of the bug/regression with instructions:

```
$myEffect = createEffect(() =>
  this.actions$.pipe(
    ofType('MY_TYPE'),
    tap(() => console.log('I am here')),
    switchMap(() => throwError('error'))
  )
)
```

The console log shows only twice.

On the first error it switches to a new observable, but without any safeguard for the next error.
https://github.com/ngrx/platform/blob/master/modules/effects/src/effects_resolver.ts#L32

## Expected behavior:

The resubscribe should work multiple times in case error happens in the stream
The error handling should be in a recursive function, something like this:

```
const handleErrors = obs$ => obs$.pipe(catchError(error => {
  if (errorHandler) errorHandler.handleError(error);
  // Return observable that produces this particular effect
  return handleErrors(obs$);
}));

const resubscribable$ = resubscribeOnError ? handleErrors(observable$) : observable$;
```

## Versions of NgRx, Angular, Node, affected browser(s) and operating system(s):

8.1.0

## I would be willing to submit a PR to fix this issue

[X] Yes (Assistance is provided if you need help submitting a pull request)
[ ] No

**Comment by timdeschryver** [comment]
Do you want to create a PR for this @peterreisz ?

**Comment by peterreisz** [comment]
Yes, done.

**Comment by peterreisz** [comment]
> Thanks for discovering and fixing the issue @peterreisz !

@alex-okrushko Your welcome, but be aware the fix was removed because of some ci issue. It will be fixed in #2165

**Comment by alex-okrushko** [comment]
@peterreisz yes, I'm aware 😃
